js/common_ajax_controller.js

```diff
@@ -0,0 +1,185 @@
+     * @returns {string}
+     */
+    const buildFormData = function (form) {
+        // Parse raw form data
+        const data = form.serializeArray();
+
+        // Try to get submit button info
+        const active_element = document.activeElement;
+        let action = null;
+        if (active_element.tagName.toLowerCase() == "button") {
+            // Form submitted by presssing a button, get value from button
+            action = $(active_element).prop('name');
+        } else {
+            // Form submitted by shift + enter, default to "update"
+            action = "update";
+        }
+
+        // Add submit button info to the form data
+        data.push({'name': '_action', 'value': action});
+
+        // Also keep action as a "direct" parameter as some internal code will
+        // look for it that way
+        data.push({'name': action, 'value': true});
+
+        // Add target itemtype
+        data.push({'name': 'itemtype', 'value': form.data('ajaxSubmitItemtype')});
+
+        return $.param(data);
+    };
+
+    /**
+     * Handle feedback message found in the reponse
+     *
+     * @param {Object} response
+     * @returns {void}
+     */
+    const handleFeedbackMessages = function (response) {
+        if (!response.messages) {
+            return;
+        }
+
+        // Display feedback messages as a toast
+        response.messages.info.forEach(message => glpi_toast_info(message));
+        response.messages.warning.forEach(message => glpi_toast_warning(message));
+        response.messages.error.forEach(message => glpi_toast_error(message));
+    };
+
+    /**
+     * Handle friendly name updates found in the reponse
+     *
+     * @param {Object} response
+     * @returns {void}
+     */
+    const handleFriendlyNameUpdate = function (response) {
+        if (!response.friendlyname || !$('#header-friendlyname').length) {
+            return;
+        }
+
+        // Update friendlyname
+        $('#header-friendlyname').text(response.friendlyname);
+    };
+
+    /**
+     * Handle thrashin status updates found in the reponse
+     *
+     * @param {Object} response
+     * @param {Object} form
+     * @returns {void}
+     */
+    const handleTrashbinStatus = function (response, form) {
+        if (response.is_deleted === undefined) {
+            return;
+        }
+
+        // Toggle "is_deleted" badge
+        if (response.is_deleted) {
+            $('#header-is-deleted').removeClass("d-none");
+            form.find('button[name=delete]').addClass("d-none");
+            form.find('button[name=purge]').removeClass("d-none");
+            form.find('button[name=restore]').removeClass("d-none");
+        } else {
+            $('#header-is-deleted').addClass("d-none");
+            form.find('button[name=delete]').removeClass("d-none");
+            form.find('button[name=purge]').addClass("d-none");
+            form.find('button[name=restore]').addClass("d-none");
+        }
```

Comment:
Cloud be simplified.
```suggestion
        $('#header-is-deleted').toggleClass("d-none", !response.is_deleted);
        form.find('button[name=delete]').toggleClass("d-none", response.is_deleted);
        form.find('button[name=purge]').toggleClass("d-none", !response.is_deleted);
        form.find('button[name=restore]').toggleClass("d-none", !response.is_deleted);
```